manifests/tpm2/ownership.pp

```diff
@@ -0,0 +1,52 @@
+# Enabling this class will set the passwords on the owner and
+# endorse hierarchies and the lock password.
+#
+# With the current version of tpm2-tools (1.7) there is no way to
+# dump the capabilities of the tpm to check if the passwords
+# are set. There for it is handled manually creating a file, owned,
+# in Puppet vardir under simp/<tpm device name>.
+#
+# NOTE: The password should be generated with passgen in order for most of the facts
+# to be functional post-ownership, as the tpm commands from tpm-tools
+# require the owner password. You can store the passwords
+# locally bit this is not recommended.
+#
+#
+# @param tpm_name     The name of the tpm in the /sys/class/tpm directory.
+#
+# @param owned        If true set the passwords, if false
+#                     It does nothing.
+#
+# One or more of the following must be set:
+# @param owner_auth   The TPM owner auth password, if '' then it will not be set.
+# @param lock_auth    The TPM lock auth password, if '' then it will not be set.
+# @param endorse_auth The TPM endorse auth password, if '' then it will not be set.
```

Comment:
I'm not sure I like the empty string logic here, but I'm not sure what to replace it with.

Reply:
I tried undef and nil and it couldn't get it to work.